api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -362,6 +366,21 @@ private Order saveOrderInternal(Order order, OrderContext orderContext) {
 				}
 			}
 		}
+
+		if (orderContext != null) {
```

Comment:
I don't think we necessarily need this restriction.

Reply:
Running validation unconditionally breaks existing behaviour/tests that never received post-validation before. Therefore we need to guard with orderContext != null so that this new validation only applies to the code paths where an OrderContext is actually provided and expected

Reply:
Are you sure?  `Sun Nov 09 15:16:07 EAT 2025` does not look the same as 1`Mon Nov 10 14:20:07 EAT 2025`,  Not sure what the "Expected" and "Actual" methods look like that.  But the assertion failure message shows differences.  Point is - if the test is failing in a situation where it should not be failing, let's figure out why and fix it and remove that unnecessary constraint on orderContext != null.

Reply:
I think that the ```orderContext != null``` check scopes the fix to the path where this inconsistency actually occurs, saveOrder(Order, OrderContext). without it, we’d start running post-validation on legacy calls that never had it, causing unrelated test failures and changing existing behavior, this keeps the fix focused on the API scenario described in the ticket, preventiing Order validation from being bypassed if certain fields are set from the OrderContext.